core/controllers/subtopic_viewer.py

```diff
@@ -18,30 +18,47 @@
 from core.controllers import acl_decorators
 from core.controllers import base
 from core.domain import subtopic_page_services
+from core.domain import topic_fetchers
 import feconf
 
 
+class SubtopicViewerPage(base.BaseHandler):
+    """Renders the subtopic viewer page."""
+
+    @acl_decorators.can_access_subtopic_viewer_page
+    def get(self, *args):
+        """Handles GET requests."""
+
+        if not constants.ENABLE_NEW_STRUCTURE_PLAYERS:
+            raise self.PageNotFoundException
+
+        self.render_template('dist/subtopic-viewer-page.mainpage.html')
+
+
 class SubtopicPageDataHandler(base.BaseHandler):
     """Manages the data that needs to be displayed to a learner on the
     subtopic page.
     """
     GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
 
     @acl_decorators.can_access_subtopic_viewer_page
-    def get(self, topic_id, subtopic_id):
+    def get(self, topic_name, subtopic_id):
```

Comment:
Just checking -- why the switch?

Reply:
Since this page is accessed from the topic viewer, which has its url as `/topic/<topicName>`, it felt right to have the url for the subtopic pages also as `/subtopic/<topicName>/<subtopicId>`